src/makeInfinityAdder.js

```diff
@@ -6,7 +6,36 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let totalSum = 0;
+  const adder = (firstNumber) => {
+    if (!firstNumber) {
+      const sumToDisplay = totalSum;
+
+      totalSum = 0;
+
+      return sumToDisplay;
+    }
+
+    totalSum = firstNumber;
+
+    function func(secondNumber) {
+      if (!arguments.length) {
+        const sumToDisplay = totalSum;
+
+        totalSum = 0;
+
+        return sumToDisplay;
+      }
+
+      totalSum += secondNumber;
+
+      return func;
+    }
+
+    return func;
+  };
```

Comment:
```suggestion
  function adder(num) {
    if (num === undefined) {
      const result = totalSum;

      totalSum = 0;

      return result;
    }

    totalSum += num;

    return adder;
  }
```

Reply:
I get it! Looks better and shorter))  Thanks!